File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Datas/Requests/TextFilterAsyncRequest.java
package toolgood.textfilter.api.Datas.Requests;

import toolgood.textfilter.api.GrpcBase.TextFilter.TextFindAllAsyncGrpcRequest;

public class TextFilterAsyncRequest {
    /**
     * 需要检测的文本
     */
    public String Txt;

    /**
     * 是否跳过 Bidi 字符，默认 false
     */
    public boolean SkipBidi;

    /**
     * 只显示位置，不显示匹配文字
     */
    public boolean OnlyPosition;

    /**
     * 请求标识，为空时会自动生成
     */
    public String RequestId;

    /**
     * 异步回调地址 可空
     */
    public String Url;

    public TextFilterAsyncRequest() {
        SkipBidi = false;
    }

    /**
     * 
     * @param txt          需要检测的文本
     * @param skipBidi     是否跳过 Bidi 字符
     * @param onlyPosition 是否只显示位置，不显示匹配文字
     * @param requestId    请求标识，为空时会自动生成
     * @param url          异步回调地址 可空
     */
    public TextFilterAsyncRequest(String txt, boolean skipBidi, boolean onlyPosition, String requestId, String url) {
        Txt = txt;
        SkipBidi = skipBidi;
        OnlyPosition = onlyPosition;
        RequestId = requestId;
        Url = url;
    }

    public TextFindAllAsyncGrpcRequest ToGrpcRequest() {
        return TextFindAllAsyncGrpcRequest.newBuilder().setTxt(Txt).setSkipBidi(SkipBidi).setOnlyPosition(OnlyPosition)
                .setUrl(Url).setRequestId(RequestId).build();
    }
}

File: api/csharp/ToolGood.TextFilter.Api/Datas/Requests/TextFilterAsyncRequest.cs
namespace ToolGood.TextFilter.Api.Datas.Requests
{
    public class TextFilterAsyncRequest
    {
        /// <summary>
        /// 需要检测的文本
        /// </summary>
        public string Txt { get; set; }

        /// <summary>
        /// 是否跳过 Bidi 字符，默认 false
        /// </summary>
        public bool SkipBidi { get; set; }
        /// <summary>
        /// 只显示位置，不显示匹配文字
        /// </summary>
        public bool OnlyPosition { get; set; }


        /// <summary>
        /// 请求标识，为空时会自动生成
        /// </summary>
        public string RequestId { get; set; }

        /// <summary>
        /// 异步回调地址 可空
        /// </summary>
        public string Url { get; set; }
        public TextFilterAsyncRequest()
        {
            SkipBidi = false;
        }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="txt">需要检测的文本</param>
        /// <param name="skipBidi">是否跳过 Bidi 字符</param>
        /// <param name="onlyPosition">是否只显示位置，不显示匹配文字</param>
        /// <param name="requestId">请求标识，为空时会自动生成</param>
        /// <param name="url">异步回调地址 可空</param>
        public TextFilterAsyncRequest(string txt, bool skipBidi = false, bool onlyPosition = false, string requestId = null, string url = null)
        {
            Txt = txt;
            SkipBidi = skipBidi;
            OnlyPosition = onlyPosition;
            RequestId = requestId;
            Url = url;
        }


#if Async
#if GRPC
        public TextFindAllAsyncGrpcRequest ToGrpcRequest()
        {
            return new TextFindAllAsyncGrpcRequest() {
                Txt = this.Txt,
                SkipBidi = this.SkipBidi,
                OnlyPosition = OnlyPosition,
                Url = this.Url,
                RequestId = this.RequestId
            };
        }
#endif
#endif

    }


}
